images/controller/internal/controllers/rvr_volume/rvr_volume_suite_test.go

```diff
@@ -0,0 +1,139 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rvrvolume_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gcustom"
+	gomegatypes "github.com/onsi/gomega/types" // cspell:words gomegatypes
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	snc "github.com/deckhouse/sds-node-configurator/api/v1alpha1"
+)
+
+func TestRvrVolume(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "RvrVolume Suite")
+}
+
+// Requeue returns a matcher that checks if reconcile result requires requeue
+func Requeue() gomegatypes.GomegaMatcher {
+	return Not(Equal(reconcile.Result{}))
+}
+
+// RequestFor creates a reconcile request for the given object
+func RequestFor(object client.Object) reconcile.Request {
+	return reconcile.Request{NamespacedName: client.ObjectKeyFromObject(object)}
+}
+
+// HaveLVMLogicalVolumeName returns a matcher that checks if RVR has the specified LLV name in status
+func HaveLVMLogicalVolumeName(llvName string) gomegatypes.GomegaMatcher {
+	if llvName == "" {
+		return SatisfyAny(
+			HaveField("Status", BeNil()),
+			HaveField("Status.LVMLogicalVolumeName", BeEmpty()),
+		)
+	}
+	return SatisfyAll(
+		HaveField("Status", Not(BeNil())),
+		HaveField("Status.LVMLogicalVolumeName", Equal(llvName)),
+	)
+}
+
+// HaveNoLVMLogicalVolumeName returns a matcher that checks if RVR has no LLV name in status
+func HaveNoLVMLogicalVolumeName() gomegatypes.GomegaMatcher {
+	return HaveLVMLogicalVolumeName("")
+}
+
+// BeLLVPhase returns a matcher that checks if LLV has the specified phase
+func BeLLVPhase(phase string) gomegatypes.GomegaMatcher {
+	if phase == "" {
+		return SatisfyAny(
+			HaveField("Status", BeNil()),
+			HaveField("Status.Phase", BeEmpty()),
+		)
+	}
+	return SatisfyAll(
+		HaveField("Status", Not(BeNil())),
+		HaveField("Status.Phase", Equal(phase)),
+	)
+}
+
+// HaveLLVWithOwnerReference returns a matcher that checks if LLV has owner reference to RVR
+func HaveLLVWithOwnerReference(rvrName string) gomegatypes.GomegaMatcher {
+	return gcustom.MakeMatcher(func(llv *snc.LVMLogicalVolume) (bool, error) {
+		ownerRef := metav1.GetControllerOf(llv)
+		if ownerRef == nil {
+			return false, nil
+		}
+		return ownerRef.Kind == "ReplicatedVolumeReplica" && ownerRef.Name == rvrName, nil
+	}).WithMessage("expected LLV to have owner reference to RVR " + rvrName)
+}
+
+// HaveFinalizer returns a matcher that checks if object has the specified finalizer
+func HaveFinalizer(finalizerName string) gomegatypes.GomegaMatcher {
+	return gcustom.MakeMatcher(func(obj client.Object) (bool, error) {
+		for _, f := range obj.GetFinalizers() {
+			if f == finalizerName {
+				return true, nil
+			}
+		}
+		return false, nil
+	}).WithMessage("expected object to have finalizer " + finalizerName)
```

Comment:
Там есть какая-то магия прикольная, но я в нее не погружался

```suggestion
	}).WithTemplate("Expected:\n{{.FormattedActual}}\n{{.To}} have finalizer:\n{{format .Data 1}}").WithTemplateData(finalizerName)
```

https://onsi.github.io/gomega/#gcustom-a-convenient-mechanism-for-building-custom-matchers